File: app/src/main/java/mechanical_man/com/analyzerdummy/mock/responses/MockExpectedGasDetectionResponse.java
package mechanical_man.com.analyzerdummy.mock.responses;

import com.mechanical_man.nst_proto.NSTProtos;

/**
 * Created by Mechanical Man on 12/11/17.
 */

public enum MockExpectedGasDetectionResponse {
    SUCCESS("success", MockExpectedGasDetection.INSTANCE.getSUCCESS());

    public final String name;
    public NSTProtos.Command response;

    MockExpectedGasDetectionResponse(String name, NSTProtos.Command response) {
        this.name = name;
        this.response = response;
    }

    @Override
    public String toString() {
        return name;
    }
}

File: app/src/main/java/mechanical_man/com/analyzerdummy/mock/responses/MockExpectedGasDetectionResponse.kt
package mechanical_man.com.analyzerdummy.mock.responses

import com.mechanical_man.nst_proto.NSTProtos

enum class MockExpectedGasDetectionResponse(
    private val testName: String,
    var response: NSTProtos.Command
) {
    SUCCESS("success", MockExpectedGasDetection.SUCCESS);

    override fun toString(): String{
        return testName
    }
}